### PR TITLE
[common] Add debezium-json format impl of DebeziumDeserializationSchema.

### PR DESCRIPTION
**Pull request by mincwang**
Add debezium-json format impl of DebeziumDeserializationSchema

### CODE REVIEW

**Comment by mincwang** [comment]
https://github.com/ververica/flink-cdc-connectors/issues/162#issue-859418217

**Comment by mincwang** [comment]
#339 https://github.com/ververica/flink-cdc-connectors/issues/162

**Comment by mincwang** [comment]
#339 https://github.com/ververica/flink-cdc-connectors/issues/162

**Comment by mincwang** [comment]
@leonardBang Can you help me review this pr ? thanks

**Comment by leonardBang** [comment]
@mincwang Thanks for the contribution, I'll find time to review this PR.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/JsonDebeziumDeserializationSchema.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * A debezium-json message format implementation of {@link DebeziumDeserializationSchema} which
+ * converts the received {@link SourceRecord} into JsonString.
+ */
+public class JsonDebeziumDeserializationSchema implements DebeziumDeserializationSchema<String> {
+    static final JsonConverter CONVERTER = new JsonConverter();
+
+    public JsonDebeziumDeserializationSchema() {
+        HashMap<String, Object> map = new HashMap<>();
+        map.put(ConverterConfig.TYPE_CONFIG, ConverterType.VALUE.getName());
+        CONVERTER.configure(map);
```
Comment (leonardBang) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/JsonDebeziumDeserializationSchema.java:42]: Do we need the schema field in DataStream applications? If not we can use  `JsonConverterConfig.schemasEnabled` to ignore the schema fields.

```json
{
    "schema": {
        "type": "struct",
        "fields": [
            {
                "type": "struct",
                "fields": [
                    {
                        "type": "int32",
                        "optional": false,
                        "field": "id"
                    },
                    {
                        "type": "string",
                        "optional": false,
                        "default": "flink",
                        "field": "name"
                    },
                    {
                        "type": "string",
                        "optional": true,
                        "field": "description"
                    },
                    {
                        "type": "double",
                        "optional": true,
                        "field": "weight"
                    }
                ],
                "optional": true,
                "name": "mysql_binlog_source.inventory_1y2uy2i.products.Value",
                "field": "before"
            },
            {
                "type": "struct",
                "fields": [
                    {
                        "type": "int32",
                        "optional": false,
                        "field": "id"
                    },
                    {
                        "type": "string",
                        "optional": false,
                        "default": "flink",
                        "field": "name"
                    },
                    {
                        "type": "string",
                        "optional": true,
                        "field": "description"
                    },
                    {
                        "type": "double",
                        "optional": true,
                        "field": "weight"
                    }
                ],
                "optional": true,
                "name": "mysql_binlog_source.inventory_1y2uy2i.products.Value",
                "field": "after"
            },
            {
                "type": "struct",
                "fields": [
                    {
                        "type": "string",
                        "optional": false,
                        "field": "version"
                    },
                    {
                        "type": "string",
                        "optional": false,
                        "field": "connector"
                    },
                    {
                        "type": "string",
                        "optional": false,
                        "field": "name"
                    },
                    {
                        "type": "int64",
                        "optional": false,
                        "field": "ts_ms"
                    },
                    {
                        "type": "string",
                        "optional": true,
                        "name": "io.debezium.data.Enum",
                        "version": 1,
                        "parameters": {
                            "allowed": "true,last,false"
                        },
                        "default": "false",
                        "field": "snapshot"
                    },
                    {
                        "type": "string",
                        "optional": false,
                        "field": "db"
                    },
                    {
                        "type": "string",
                        "optional": true,
                        "field": "sequence"
                    },
                    {
                        "type": "string",
                        "optional": true,
                        "field": "table"
                    },
                    {
                        "type": "int64",
                        "optional": false,
                        "field": "server_id"
                    },
                    {
                        "type": "string",
                        "optional": true,
                        "field": "gtid"
                    },
                    {
                        "type": "string",
                        "optional": false,
                        "field": "file"
                    },
                    {
                        "type": "int64",
                        "optional": false,
                        "field": "pos"
                    },
                    {
                        "type": "int32",
                        "optional": false,
                        "field": "row"
                    },
                    {
                        "type": "int64",
                        "optional": true,
                        "field": "thread"
                    },
                    {
                        "type": "string",
                        "optional": true,
                        "field": "query"
                    }
                ],
                "optional": false,
                "name": "io.debezium.connector.mysql.Source",
                "field": "source"
            },
            {
                "type": "string",
                "optional": false,
                "field": "op"
            },
            {
                "type": "int64",
                "optional": true,
                "field": "ts_ms"
            },
            {
                "type": "struct",
                "fields": [
                    {
                        "type": "string",
                        "optional": false,
                        "field": "id"
                    },
                    {
                        "type": "int64",
                        "optional": false,
                        "field": "total_order"
                    },
                    {
                        "type": "int64",
                        "optional": false,
                        "field": "data_collection_order"
                    }
                ],
                "optional": true,
                "field": "transaction"
            }
        ],
        "optional": false,
        "name": "mysql_binlog_source.inventory_1y2uy2i.products.Envelope"
    }
}
```

File: flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/MySqlSourceITCase.java [hunk 2]
```diff
@@ -54,4 +55,26 @@ public void testConsumingAllEvents() throws Exception {
 
         env.execute("Print MySQL Snapshot + Binlog");
     }
+
+    @Test
+    public void testConsumingAllEventsWithJsonFormat() throws Exception {
```
Comment (leonardBang) [inline flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/MySqlSourceITCase.java:60]: The ITCase is a little having and the test is set to `@ingore`, could you add a unit test to check all data types?

**Review by leonardBang — COMMENTED** [review]
Thanks @mincwang for the great work, I left some comments

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/JsonDebeziumDeserializationSchema.java (hunk 1 above)
Comment (mincwang) [reply flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/JsonDebeziumDeserializationSchema.java:42]: Ok, i think that is good advice.

File: flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/MySqlSourceITCase.java (hunk 2 above)
Comment (mincwang) [reply flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/MySqlSourceITCase.java:60]: I will refer to this `MySqlConnectorITCase#testAllTypes` to complete the ITCase.

File: flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/JsonDebeziumDeserializationSchema.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    public JsonDebeziumDeserializationSchema(boolean includeSchema) {
+        config(includeSchema);
+    }
+
+    private void config(boolean includeSchema) {
+        HashMap<String, Object> configs = new HashMap<>();
+        configs.put(ConverterConfig.TYPE_CONFIG, ConverterType.VALUE.getName());
+        configs.put(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, includeSchema);
+        CONVERTER.configure(configs);
+    }
```
Comment (leonardBang) [inline flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/JsonDebeziumDeserializationSchema.java:53]: minor， could simplify to 
```
public JsonDebeziumDeserializationSchema() {
        this(false);
    }

    public JsonDebeziumDeserializationSchema(boolean includeSchema) {
        HashMap<String, Object> configs = new HashMap<>();
        configs.put(ConverterConfig.TYPE_CONFIG, ConverterType.VALUE.getName());
        configs.put(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, includeSchema);
        CONVERTER.configure(configs);
    }
```

File: flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/MySqlSourceITCase.java [hunk 4]
```diff
@@ -54,4 +78,131 @@ public void testConsumingAllEvents() throws Exception {
...
+        while (!iterator.hasNext()) {
+            Thread.sleep(100);
+        }
+    }
+
+    private static List<String> readLines(String resource) throws IOException {
+        final URL url = MySqlSourceITCase.class.getClassLoader().getResource(resource);
+        assert url != null;
+        java.nio.file.Path path = new File(url.getFile()).toPath();
+        return Files.readAllLines(path);
```
Comment (leonardBang) [inline flink-connector-mysql-cdc/src/test/java/com/ververica/cdc/connectors/mysql/MySqlSourceITCase.java:189]: we could avoid use `assert`
```
      Paths.get(
                Objects.requireNonNull(MySqlSourceITCase.class.getClassLoader().getResource("."))
                        .toURI());
```

File: flink-connector-mysql-cdc/src/test/resources/file/debezium-data-schema-exclude.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+{"before":null,"after":{"id":1,"tiny_c":127,"tiny_un_c":255,"small_c":32767,"small_un_c":65535,"int_c":2147483647,"int_un_c":4294967295,"int11_c":2147483647,"big_c":9223372036854775807,"varchar_c":"Hello World","char_c":"abc","float_c":123.10199737548828,"double_c":404.4443,"decimal_c":"EtaH","numeric_c":"AVo=","boolean_c":1,"date_c":18460,"time_c":64822000000,"datetime3_c":1595008822123,"datetime6_c":1595008822123456,"timestamp_c":"2020-07-17T18:00:22Z","file_uuid":"ZRrtCDkPSJOy8TaSPnt0AA=="},"op":"r","transaction":null}
```
Comment (leonardBang) [inline flink-connector-mysql-cdc/src/test/resources/file/debezium-data-schema-exclude.txt:1]: Nice change, could we make the format more readable as following, we can then use `JSONObject.parseObject("debezium-data-schema-exclude.json").getJSONObject("expect_line_1")`
```json
{
    "expect_line_1": {
        "before": null,
        "after": {
            "id": 1
        },
        "op": "r",
        "transaction": null
    },
    "expect_line_2": {
        "before": {
            "id": 1
        },
        "after": {
            "id": 1
        },
        "op": "u",
        "transaction": null
    }
}
```
And you may need to add the  `flink-connector-mysql-cdc/src/test/resources/file/*.json` to line 187 of `pom.xml` files to checkstyle.

**Review by leonardBang — APPROVED** [review]
Thanks @mincwang for the great work, LGTM
I append one commit for minor tests improvement, will merge this once CI passed

**Comment by leonardBang** [comment]
This closes #339 #162

**Comment by mincwang** [comment]
Thanks for @leonardBang  review and suggestion optimization. I also learned a lot from it.

**Pull request merged**